### ISSUE TITLE
Checking of user-defined-operations

### ISSUE THREAD

**Issue by JonathanStarup** [flix/flix#7790]
The parser expects user defined operations to be binary but this is not enforced
```
def <><(x: Int32): Int32 = x
def main(): Int32 = <>< 42
```

**Comment by magnus-madsen** [comment]
Where is the error here? Do you want the weeder to reject the fish because it consists only of symbols and is non-binary?

**Comment by herluf-ba** [comment]
I am getting this on `master`
```text
-- Parse Error (OtherExpr) -------------------------------------------------- main/foo.flix

>> Expected <expression> before <user-defined operator>.

2 | def main(): Int32 = <>< 42
                        ^^^
                        Here
```
So the parser rejects unary user-defined operators. 

It's not difficult I think to support them though... I'm not sure what we want here? 😅

**Comment by magnus-madsen** [comment]
I don't think we want to support them...

**Comment by JonathanStarup** [comment]
Yeah i think thats fine, but we should blame the definition not the use site.

I don't know which phase, but something should point to the `<><` definition an say that its wrong to have !=2 arguments

**Comment by herluf-ba** [comment]
This seems like a `Weeder` error to me.
If the name in a def is a user-operation it should check that there are exactly 2 formal parameters.

**Comment by magnus-madsen** [comment]
Ok. I agree with that. Herluf could I ask you to prepare a fix?

**Comment by magnus-madsen** [comment]
Low priority.
